src/expr.rs

```diff
@@ -16,34 +18,143 @@ pub enum Expr {
     Not(Box<Expr>),
     Any(Vec<Expr>),
     All(Vec<Expr>),
+    MinVer(Bound),
+}
+
+pub enum ExprError {
+}
+
+impl ExprError {
+    fn nightly_date(mindate: Date, date: Date) -> Self {
+        unimplemented!()
+    }
+
+    fn nightly_channel(channel: Channel) -> Self {
+        unimplemented!()
+    }
+
+    fn nightly_release(release: Release) -> Self {
+        unimplemented!()
+    }
+
+    fn release(minrel: Release, release: Release) -> Self {
+        unimplemented!()
+    }
+
+    fn bad_bound(minver: Bound, bound: Bound) -> Self {
+        unimplemented!()
+    }
+
+    fn duplicate_minver(minver: Bound, new_minver: Bound) -> Self {
+        unimplemented!()
+    }
+}
+
+pub type ExprResult<T> = std::result::Result<T, ExprError>;
+
+lazy_static! {
+    static ref MINVER: RwLock<Option<Bound>> = RwLock::new(None);
```

Comment:
It looks like you are planning to rely on invocation order and sharing data through a static in the proc macro's process. Be aware that the compiler makes no guarantee about what order macro invocations run in or whether different macro calls all run in the same process, so I don't plan to accept an implementation like this.

Reply:
I thought that might be a problem, but documentation seems to be thin. Is there a place to store stateful information like this? I assume so since I'm not sure how/where `clippy` would store warning state information, but maybe that's a plugin thing and not available to proc macros?

Reply:
I don't know of an established place to store information across proc macro calls without putting it in the generated code. Clippy is not implemented as a proc macro.

Reply:
Well, if it does go into the generated code, how would a proc macro access it anyways? Maybe this isn't possible with a proc macro today?